src/app/services/game-play.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, fromEvent } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { AnswerService } from './answer.service';
+import { Answer } from '../models/answer';
+
+interface Game {
+  answer: Answer;
+  acceptedGuesses: string[];
+  status: Status;
+  currentGuessIdx: number;
+  currentGuess: string;
+};
+
+enum Status {
+  playing,
+  won,
+  lost
+};
+
+const emptyGame: Game = {
+  answer: <Answer>{},
```

Comment:
I tend to make fields nullable instead of casting them. That way you won't accidentally try to get `game.answer.word` because you'll know you might not even have a `game.answer`

Reply:
Oh I love that idea, thank you!

Reply:
If I was to do this with the answer example you highlighted above, would you make answer optional using the `?` operator? Or would it be better to make answer a nullable type such as:
```
type Nullable<T> = T | null;

interface Game {
  answer: Nullable<Answer>;
  ...
};
```

Reply:
I think it makes more sense to use `?`:

```Typescript
interface Game {
  answer?: Answer;
  // OR
  answer: Answer | null;
};
```

There is a subtle difference between those two, the first one means `Answer` could be `undefined`, the second one means it could be `null`.